services/src/main/java/org/keycloak/authentication/authenticators/x509/CertificateValidator.java

```diff
@@ -675,6 +682,21 @@ public GotCRLRelativePath cRLLoader(CRLLoaderImpl cRLLoader) {
             }
 
             public class GotOCSP {
+                public GotOCSP oCSPResponseCertificate(String responderCert) {
+                    if (responderCert != null && !responderCert.isEmpty()) {
+                        try {
+                            _responderCert = XMLSignatureUtil.getX509CertificateFromKeyInfoString(responderCert);
```

Comment:
I wonder what is the performance impact of this? I think it will be good to have a possibility to "cache" the X509Certificate object, so it doesn't need to be parsed from PEM on every request. However this is probably not for this PR, but rather an improvement for the future. Just mentioning it in this PR when thinking about it :)